### PR TITLE
Feature/SK-1071 | Initial refactor of client

### PR DESCRIPTION
**Pull request by niklastheman**
### fedn client start-v2 command added

OBS this PR does not replace the current client, it's and initial step to a new client (refactor)

Mainly tried to clean up the existing client. The existing client is left as is. The idea is to let this new solution exist in parallel until it can be the main solution. The functionality of the client is divided into three main classes, GrpcHandler, ClientApi and the "new" Client class. The Client class implements the ClientApi. The Client class can be understood as an example of how to set up your own client.

### CODE REVIEW

**Review by Wrede — APPROVED** [review]
I want @ahellander to look at this also

File: fedn/network/clients/client_api.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        self.package = package
+        self.ip = ip
+        self.port = port
+        self.helper_type = helper_type
+
+
+# Enum for respresenting the result of connecting to the FEDn API
+class ConnectToApiResult(enum.Enum):
+    Assigned = 0
+    ComputePackgeMissing = 1
```
Comment (ahellander) [inline fedn/network/clients/client_api.py:31]: Spelling error .

File: fedn/network/clients/client_api.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        self.grpc_handler: GrpcHandler = None
+
+    def subscribe(self, event_type: str, callback: callable):
+        """Subscribe to a specific event."""
+        if event_type in self._subscribers:
+            self._subscribers[event_type].append(callback)
+        else:
+            raise ValueError(f"Unsupported event type: {event_type}")
+
+    def notify_subscribers(self, event_type: str, *args, **kwargs):
```
Comment (ahellander) [inline fedn/network/clients/client_api.py:69]: I don't understand, what subscribers might a client have?

**Review by ahellander — APPROVED** [review]
Looks like a good refactor, I added two minor comments, but I'm sure this will be iterated on a number of times before replacing v1.

File: fedn/network/clients/client_api.py (hunk 2 above)
Comment (niklastheman) [reply fedn/network/clients/client_api.py:69]: @ahellander if you look in the client_v2.py file you can see how it is used. The overall idea is that client_api exposes functionality that a client can then make use of... In this particular example the client subscribes to events raised from the client_api. Events include training, validation or inference.

Comment (niklastheman) [reply fedn/network/clients/client_api.py:69]: I will add a little example too.

File: fedn/network/clients/client_api.py (hunk 1 above)
Comment (niklastheman) [reply fedn/network/clients/client_api.py:31]: Will fix this.

**Pull request merged**